wasm_exec/internal/compiler_test.go

```diff
@@ -0,0 +1,166 @@
+package internal_test
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/tetratelabs/wazero"
+	"github.com/tetratelabs/wazero/api"
+	. "github.com/tetratelabs/wazero/experimental"
+	"github.com/tetratelabs/wazero/internal/testing/require"
+	"github.com/tetratelabs/wazero/internal/wasm"
+	"github.com/tetratelabs/wazero/internal/wasm/binary"
+	"github.com/tetratelabs/wazero/wasm_exec"
+	. "github.com/tetratelabs/wazero/wasm_exec/internal"
+)
+
+// testCtx is an arbitrary, non-default context. Non-nil also prevents linter errors.
+var testCtx = context.WithValue(context.Background(), struct{}{}, "arbitrary")
+
+// Test_compileJsWasm ensures the infrastructure to generate wasm on-demand works.
+func Test_compileJsWasm(t *testing.T) {
```

Comment:
can this work on scratch containers?

Reply:
no that's why there's some logic later to skip the test. scratch doesn't have the "go" command.

Reply:
as this isn't a compiler per-se we shouldn't need to run also on scratch anyway. the main thing is to use multiple versions of go. This way, we can check the output wasm in multiple versions, which ideally aren't terribly different per os, or different at all!

Reply:
ps in the example test, I simply bail out when go compile fails. the instructions tell users to do it.